extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala

```diff
@@ -414,6 +415,10 @@ trait LineageParser {
             })
         }
 
+      case p: LineagePermanentViewMarker =>
```

Comment:
I'm not sure why it is so complex. can not just `case v: View if !v.isTemp && enableParsePermanentView` ?

Reply:
Because we are using optimizedPlan, `select * from test_view` will resolve the view table into actual SQL.

Reply:
> It seems overkill to me that people need to inject SQL extension to make lineage working. The extension may break their SQL job, e.g., conflict with auth extension.
> 
> Is there a simple way to workaround ?

Yes, it may affect other plugins. I'll try other ways again.

Reply:
Replaced `optimizedPlan` with `analyzedPlan`.

Reply:
We should take care of other logical plan which is eliminated at optimizer side. e.g., `SubqueryAlias`

Reply:
There should be no problem here. The entire parsing logic was previously based on optimization, and after optimization, it can also be handled.

Reply:
can we add some tests to confirm it ? e.g.,
```
create table t (c1 int, c2 string) using parquet;
select tmp.c1, c2 from (select * from t) tmp;
```

Reply:
This type of case already exists in UT：
https://github.com/apache/kyuubi/blob/38cf59d47be8f6d1b45562259c9e6342215888de/extensions/spark/kyuubi-spark-lineage/src/test/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParserHelperSuite.scala#L946-L965

Reply:
hmm, seems not exact same. The case I want to ensure is the qulified name refernce a subqery alias. That means, the key is `select tmp.c1 from (select * from t) tmp` rather than `select c1 from (select * from t) tmp`

Reply:
I see

Reply:
```
Project [a#2, b#3]
+- SubqueryAlias tmp
   +- Project [a#2, b#3, c#4]
      +- SubqueryAlias spark_catalog.default.table1
         +- HiveTableRelation [`default`.`table1`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#2, b#3, c#4], Partition Cols: []]
```
The test is fine because the subqueryAlias will be skipped during parsing. I will add this test case.